proxy.js

```diff
@@ -0,0 +1,71 @@
+'use strict';
+
+const processFn = (fn, opts) => function () {
+	const that = this;
+	const P = opts.promiseModule;
+	const args = new Array(arguments.length);
+
+	for (let i = 0; i < arguments.length; i++) {
+		args[i] = arguments[i];
+	}
```

Comment:
Could you also integrate https://github.com/sindresorhus/pify/commit/65d5fc9b14e7778871b189d65c07316273014c0c? Would like to remove this boilerplate code.